File: test-suite/ruby/suppression/suppression_buggy.rb
# frozen_string_literal: false
#
# GH #91 suppression fixture (bead A7) for the Ruby module, twin of
# suppression_buggy_nomarkers.rb (identical buggy code, markers removed,
# except where noted: the ruby lifecycle helper is marker-blind, so the
# markered twin pairs the file handle through the block form).
# Every native ruby finding below (eval, single-string system, File.open
# without close) carries an ubs:ignore marker in one of the documented
# arrangements, so scanning this file must report zero findings while the
# nomarkers twin reproduces them.

module SuppressionProbe
  SPOOL_PATH = 'spool.txt'

  # Arrangement 1: previous-line marker. The module count filter is
  # same-line only, so the trailing twin rides along; the postprocess
  # engine (GH #91) covers the previous-line placement.
  def self.eval_prev_line(code)
    # ubs:ignore -- fixture: marker on the line immediately above the finding
    eval(code) # ubs:ignore -- fixture: trailing twin for the module count filter
    code
  end

  # Arrangement 2: trailing marker on the flagged line itself.
  def self.eval_trailing(code)
    eval(code) # ubs:ignore -- fixture: trailing marker
    code
  end

  # Arrangement 2b: rule-scoped trailing marker (ubs:ignore[rule]).
  def self.eval_rule_scoped(code)
    eval(code) # ubs:ignore[rb.eval-exec] -- fixture: rule-scoped trailing marker
    code
  end

  # Arrangement 3: multi-line statement, marker on the flagged
  # continuation line of that statement.
  def self.system_multiline(cmd)
    outcome =
      system("sh -c \"#{cmd}\"") # ubs:ignore -- fixture: marker on a physical line of a multi-line statement
    outcome == true
  end

  # Arrangement 4: formatter-relocated marker on the first line inside
  # the block whose opening line the scanner flags. The nomarkers twin
  # holds the handle outside a block (never closed), which is the finding
  # this arrangement hosts.
  def self.write_spool
    File.open(SPOOL_PATH, 'w') do |spool|
      # ubs:ignore -- fixture: formatter moved the trailing marker inside the block
      spool.write('spooled')
    end
  end
end

File: test-suite/swift/suppression/suppression_buggy.swift
// GH #91 suppression fixture (bead A7) for the Swift module, twin of
// suppression_buggy_nomarkers.swift (identical buggy code, no markers).
// Force-try, force-cast and Timer lifecycle findings below carry markers
// in the documented arrangements. The unmarked print calls remain positive
// informational controls, including the calls after previous-line scopes.
// Scanning this file must omit the marked hazard sites while retaining
// those print findings; the nomarkers twin also reproduces the hazards.

import Foundation

final class SuppressionFixture {

    private var timer: Timer?

    // Arrangement 1: previous-line marker.
    func decodePrevLine(_ payload: Data) throws {
        // ubs:ignore -- fixture: marker on the line immediately above the finding
        let decoded = try! JSONDecoder().decode([String: Int].self, from: payload)
        print(decoded)
    }

    // Arrangement 2: trailing marker on the flagged line itself.
    func castTrailing(_ any: Any) {
        let label = any as! String // ubs:ignore -- fixture: trailing marker
        print(label)
    }

    // Arrangement 3: multi-line statement, marker on a continuation line.
    func decodeMultiline(_ payload: Data) throws {
        let decoded = try! JSONDecoder().decode(
            [String: Int].self, from: payload) // ubs:ignore -- fixture: marker on a physical line of a multi-line statement
        print(decoded)
    }

    // Arrangement 4: formatter-relocated marker on the first line inside a block.
    func startTimer() {
        timer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: true) { _ in
            // ubs:ignore -- fixture: formatter moved the marker inside the block
            print("tick")
        }
    }

    // Arrangement 5: scopes name public diagnostic identifiers explicitly.
    // Each scope suppresses only its named force-try or force-cast finding.
    // Unmarked print calls continue to exercise independent diagnostics.
    func ruleScopedPrevLine(_ payload: Data) throws {
        // ubs:ignore[swift.force-try,swift.optionals.try-bang,swift.optionals.force-some] -- fixture: rule-scoped marker above the finding
        let decoded = try! JSONDecoder().decode([String: Int].self, from: payload)
        print(decoded)
    }

    func ruleScopedTrailing(_ any: Any) {
        let flag = any as! Int // ubs:ignore[swift.force-cast,swift.optionals.as-bang,swift.optionals.force-some] -- fixture: rule-scoped trailing marker
        print(flag)
    }
}
